fil_fungible_token/src/token/mod.rs

```diff
@@ -535,12 +548,38 @@ mod test {
         assert_eq!(last_called, expected);
     }
 
+    #[test]
+    fn it_wraps_a_previously_loaded_state_tree() {
+        struct ActorState {
+            token_state: TokenState,
+        }
+
+        // simulate the token state being a node in a larger state tree
+        let bs = MemoryBlockstore::default();
+        let actor_state = ActorState { token_state: TokenState::new(&bs).unwrap() };
+
+        // wrap the token state, moving it into a TokenHandle
+        let mut token = Token::new(
+            &bs,
+            FakeMessenger::new(TOKEN_ACTOR.id().unwrap(), 6),
+            actor_state.token_state,
+        );
+
+        token.mint(TOKEN_ACTOR, TREASURY, &TokenAmount::from(1), &Default::default()).unwrap();
+        // gets a read-only state
+        let state = token.borrow_state();
+        assert_eq!(state.supply, TokenAmount::from(1));
+
+        // actor_state.token_state is moved when `wrap` is called so cannot be accessed anymore
+        // assert_eq!(actor_state.token_state.supply, TokenAmount::zero());
```

Comment:
But did it write back to actor_state or not? If I now want to persist that state, I need to be able to access actor_state to serialize it.